Updating package dependencies to fix DoS vulnerability

acolombier opened the pull request:
Update dependency `js-yaml` to fix medium security issue. Snyk report [here](https://snyk.io/vuln/SNYK-JS-JSYAML-173999)

martysweet commented on package.json:
```diff
@@ -39,10 +38,11 @@
-    "typescript": "~2.6.2",
     "dependency-check": "^2.9.1",
-    "proxyquire-2": "^1.0.7"
+    "js-yaml": "^3.13.0",
```
This should be a dependency, not dev dependency.

acolombier commented:
Hi @martysweet 

Sorry about the mistake. I fixed it and also had to lock `mocha` to version 5 since they droped support of Node 4 LTS. I also updated the `CHANGELOG` so it hopefully save you some time! :)

Antoine

martysweet commented:
Thanks @acolombier, this will be released shortly.
